bindings/java/src/org/sleuthkit/datamodel/CaseDatabaseFactory.java

```diff
@@ -0,0 +1,595 @@
+		this.dbCreationHelper = new PostgreSQLDbCreationHelper(caseName, info);
+	}
+	
+	/**
+	 * Creates and initializes the case database.
+	 * Currently the case must be reopened after creation.
+	 * 
+	 * @throws TskCoreException 
+	 */
+	void createCaseDatabase() throws TskCoreException {
+		createDatabase();
+		initializeSchema();
+	}
+	
+	/**
+	 * Create the database itself (if necessary)
+	 * 
+	 * @throws TskCoreException 
+	 */
+	private void createDatabase() throws TskCoreException {
+		dbCreationHelper.createDatabase();
+	}
+	
+	/**
+	 * Initialize the database schema
+	 * 
+	 * @throws TskCoreException 
+	 */
+	private void initializeSchema() throws TskCoreException {
+		try (Connection conn = dbCreationHelper.getConnection()) {
+			// Perform any needed steps before creating the tables
+			dbCreationHelper.performPreInitialization(conn);
+
+			// Add schema version
+			addDbInfo(conn);
+
+			// Add tables
+			addTables(conn);
+			dbCreationHelper.performPostTableInitialization(conn);
+		
+			// Add indexes
+			addIndexes(conn);
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error initializing case database", ex);
+		}
+	}
+	
+	/**
+	 * Create and populate the db_info tables
+	 * 
+	 * @param conn the database connection
+	 * 
+	 * @throws TskCoreException 
+	 */
+	private void addDbInfo(Connection conn) throws TskCoreException {
+		CaseDbSchemaVersionNumber version = SleuthkitCase.CURRENT_DB_SCHEMA_VERSION;
+		long tskVersionNum = SleuthkitJNI.getSleuthkitVersion(); // This is the current version of TSK
+		
+		try (Statement stmt = conn.createStatement()) {
+			stmt.execute("CREATE TABLE tsk_db_info (schema_ver INTEGER, tsk_ver INTEGER, schema_minor_ver INTEGER)");
+			stmt.execute("INSERT INTO tsk_db_info (schema_ver, tsk_ver, schema_minor_ver) VALUES (" + 
+					version.getMajor() + ", " + tskVersionNum + ", " + version.getMinor() + ");");
+
+			stmt.execute("CREATE TABLE tsk_db_info_extended (name TEXT PRIMARY KEY, value TEXT NOT NULL);");
+			stmt.execute("INSERT INTO tsk_db_info_extended (name, value) VALUES ('TSK_VERSION', '" + tskVersionNum + "');");
+			stmt.execute("INSERT INTO tsk_db_info_extended (name, value) VALUES ('SCHEMA_MAJOR_VERSION', '" + version.getMajor() + "');");
+			stmt.execute("INSERT INTO tsk_db_info_extended (name, value) VALUES ('SCHEMA_MINOR_VERSION', '" + version.getMinor() + "');");
+			stmt.execute("INSERT INTO tsk_db_info_extended (name, value) VALUES ('CREATION_SCHEMA_MAJOR_VERSION', '" + version.getMajor() + "');");
+			stmt.execute("INSERT INTO tsk_db_info_extended (name, value) VALUES ('CREATION_SCHEMA_MINOR_VERSION', '" + version.getMinor() + "');");
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error initializing db_info tables", ex);
+		}
+	}
+	
+	/**
+	 * Add and initialize the database tables 
+	 * 
+	 * @param conn the database connection
+	 * 
+	 * @throws TskCoreException 
+	 */
+	private void addTables(Connection conn) throws TskCoreException {
```

Comment:
I see Codacy's point regarding the length of this method. How about breaking it up into some smaller methods?